honeybadgermpc/reed_solomon.py

```
@@ -0,0 +1,381 @@
+                self._guess_encoded = None
+                self._optimistic = False
+
+        if success and len(self._available_points) >= self._min_points_required():
+            # Decoding successful
+            self._result = self._guess_decoded
+
+        return success
+
+    def _robust_update(self):
+        while self._num_decoded < self.batch_size:
+            print(f"len(z): {len(self._z)} len(data): {len(self._available_data[0])}")
+            decoded, errors = self.robust_decoder.robust_decode(self._z,
+                                                                self._available_data[0])
+            # Need to wait for more data
+            if decoded is None:
+                break
+
+            num_agreement = len(self._available_points) - len(errors)
+            if num_agreement >= self._min_points_required():
+                # Errors detected considered to be confirmed errors
+                self._confirmed_errors |= set(errors)
+                self._available_points -= set(errors)
+                error_indices = []
+                for e in errors:
+                    error_idx = self._z.index(e)
+                    error_indices.append(error_idx)
+                    del self._z[error_idx]
+                self._num_decoded += 1
+                self._available_data = self._available_data[1:]
+                self._partial_result.append(decoded)
+
+                for error_idx in error_indices:
+                    for i in range(len(self._available_data)):
+                        del self._available_data[i][error_idx]
+            else:
+                break
+
+        # We're done
+        if self._num_decoded == self.batch_size:
+            self._result = self._partial_result
+
+    # Public API
+    def add(self, idx, data):
+        if self.done() or idx in self._available_points or \
+                idx in self._confirmed_errors:
+            return
+
+        if self._validate(data) is False:
+            return
```

if validation fails, do we want to raise an error?